phases/00-setup-and-tooling/04-apis-and-keys/code/first_api_call_openrouter.py

```
@@ -0,0 +1,76 @@
+        "Content-Type": "application/json",
+        "Authorization": f"Bearer {api_key}",
+    }
+    body = json.dumps(
+        {
+            "model": MODEL,
+            "max_tokens": 256,
+            "messages": [{"role": "user", "content": PROMPT}],
+        }
+    ).encode()
+
+    req = urllib.request.Request(URL, data=body, headers=headers, method="POST")
```

Ruff (0.15.12)

[error] Audit URL open for permitted schemes. Allowing use of `file:` or custom schemes is often unexpected.

(S310)